columnflow/tasks/framework/mixins.py

```diff
@@ -691,10 +691,35 @@ def store_parts(self) -> law.util.InsertableDict:
         # since MLTraining is no CalibratorsMixin, SelectorMixin, ProducerMixin, ConfigTask,
         # all these parts are missing in the `store_parts`
 
-        configs_repr = "__".join(self.configs)
+        configs_repr = "__".join(self.configs[:5])
+
+        if len(self.configs) > 5:
+            configs_repr += f"_{law.util.create_hash(self.configs[5:])}"
+
         parts.insert_after("task_family", "configs", configs_repr)
 
-        # TODO: parts for calib, sel, prod
+        task_array_functions = {
+            "calib": self.calibrators,
+            "sel": tuple((sel,) for sel in self.selectors),
+            "prod": self.producers,
+        }
+
+        for label, fct_names in task_array_functions.items():
+            if all(map(lambda x: x == fct_names[0], fct_names)):
+                # when functions are the same per config, only use them once
+                fct_names = fct_names[0]
+                n_fct_per_config = str(len(fct_names))
+            else:
+                # when functions differ between configs, flatten
+                n_fct_per_config = "".join(str(len(x)) for x in fct_names)
+                fct_names = sum(fct_names[1:], fct_names[0])
+
+            part = "_".join(fct_names[:2])
```

Comment:
For consistency with other mixing, the joiner should probably be a double underscore here.
```suggestion
            part = "__".join(fct_names[:2])
```